### PR TITLE
Merge 1 4 into master

### PR DESCRIPTION
**Pull request by jmleroux**

### CODE REVIEW

File: features/Context/Page/Base/ProductEditForm.php [hunk 1]
```diff
@@ -40,35 +46,83 @@ public function __construct($session, $pageFactory, $parameters = [])
...
-        }
+        // Open select2
+        $selector->click();
 
-        $options = $this->spin(function () use ($list) {
-            return $list->findAll('css', 'li');
-        }, 'No attributes found in available attributes list');
+        $list = $this->spin(function () {
+            return $this->getElement('Available attributes list');
+        }, sprintf('Cannot find the attribute list element'));
```
Comment (pierallard) [inline features/Context/Page/Base/ProductEditForm.php]: This sprintf is unused

Comment (grena) [inline features/Context/Page/Base/ProductEditForm.php]: Right, I'll update this ;)

File: spec/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizerSpec.php [hunk 2]
```diff
@@ -37,10 +45,12 @@ public function it_adds_the_attribute_id_to_the_normalized_attribute($normalizer
         $price->getDefaultMetricUnit()->willReturn('');
         $price->getMaxFileSize()->willReturn('');
         $price->getSortOrder()->willReturn(2);
+        $price->getGroup()->willReturn(null);
 
         $fieldProvider->getField($price)->willReturn('akeneo-text-field');
         $emptyValueProvider->getEmptyValue($price)->willReturn([]);
 
+
```
Comment (jmleroux) [inline spec/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizerSpec.php]: extra blank line

File: src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeSearchableRepository.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    /**
+     * @param string $search
+     * @param array  $options
+     *
+     * @return QueryBuilder
+     */
+    protected function findBySearchQb($search, array $options)
+    {
+        //TODO: refactor on master because this is exactly the same that FamilySearchableRepository
+        //TODO: and should be put in Akeneo\Bundle\StorageUtilsBundle\Doctrine\ORM\Repository\SearchableRepository
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeSearchableRepository.php]: just a reminder ;)

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php [hunk 4]
```diff
@@ -23,44 +26,98 @@ class AttributeController
...
+     * @param ObjectFilterInterface         $attributeFilter
+     * @param SearchableRepositoryInterface $attributeSearchRepository
      */
     public function __construct(
         AttributeRepositoryInterface $attributeRepository,
         NormalizerInterface $normalizer,
-        CollectionFilterInterface $collectionFilter
+        TokenStorageInterface $tokenStorage,
+        ObjectFilterInterface $attributeFilter,
+        SearchableRepositoryInterface $attributeSearchRepository = null
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php]: remove `null`

**Pull request merged**